tests/contrib/django_util/test_decorators.py

```diff
@@ -0,0 +1,254 @@
+        self.assertIsNotNone(request.oauth)
+        self.assertFalse(request.oauth.has_credentials())
+
+
+class OAuth2RequiredDecoratorTest(TestWithDjangoEnvironment):
+
+    def setUp(self):
+        super(OAuth2RequiredDecoratorTest, self).setUp()
+        self.save_settings = copy.deepcopy(django.conf.settings)
+
+        if getattr(
+                django.conf.settings,
+                'GOOGLE_OAUTH2_STORAGE_MODEL', None) is not None:
+            del django.conf.settings.GOOGLE_OAUTH2_STORAGE_MODEL
+
+        reload_module(oauth2client.contrib.django_util)
+        self.user = User.objects.create_user(
+            username='bill', email='bill@example.com', password='hunter2')
+
+    def tearDown(self):
+        super(OAuth2RequiredDecoratorTest, self).tearDown()
+        django.conf.settings = copy.deepcopy(self.save_settings)
+
+    def test_redirects_without_credentials(self):
+        request = self.factory.get('/test')
+        request.session = self.session
+
+        @decorators.oauth_required
+        def test_view(request):
+            return http.HttpResponse('test')  # pragma: NO COVER
+
+        response = test_view(request)
+        self.assertIsInstance(response, http.HttpResponseRedirect)
+        self.assertEqual(parse.urlparse(response['Location']).path,
+                         '/oauth2/oauth2authorize/')
+        self.assertIn(
+            'return_url=%2Ftest', parse.urlparse(response['Location']).query)
+
+        self.assertEqual(response.status_code,
+                         http.HttpResponseRedirect.status_code)
+
+    @mock.patch('oauth2client.contrib.django_util.UserOAuth2', autospec=True)
+    def test_has_credentials_in_storage(self, UserOAuth2):
+        request = self.factory.get('/test')
+        request.session = mock.MagicMock()
+
+        @decorators.oauth_required
+        def test_view(request):
+            return http.HttpResponse("test")
+
+        my_user_oauth = mock.MagicMock()
+
+        UserOAuth2.return_value = my_user_oauth
+        my_user_oauth.has_credentials.return_value = True
+
+        response = test_view(request)
+        self.assertEqual(response.status_code, http_client.OK)
+        self.assertEqual(response.content, b"test")
+
+    @mock.patch('oauth2client.contrib.dictionary_storage.OAuth2Credentials')
+    def test_has_credentials_in_storage_no_scopes(
+            self, OAuth2Credentials):
+        request = self.factory.get('/test')
+
+        request.session = mock.MagicMock()
+        credentials_mock = mock.Mock(
+            scopes=set(django.conf.settings.GOOGLE_OAUTH2_SCOPES))
+        credentials_mock.has_scopes.return_value = False
+
+        OAuth2Credentials.from_json.return_value = credentials_mock
+
+        @decorators.oauth_required
+        def test_view(request):
+            return http.HttpResponse("test")  # pragma: NO COVER
+
+        response = test_view(request)
+        self.assertEquals(
```

Comment:
Always leave the optional ess off the end of `assertEqual` (here and in one other location in this module, but I think that's finally it for this change!).